src/core/imageGallery/Lightbox.tsx

```diff
@@ -0,0 +1,162 @@
+          lastElement.focus();
+        } else if (!event.shiftKey && document.activeElement === lastElement) {
+          event.preventDefault();
+          focusableElements[0].focus();
+        }
+      }
+    };
+
+    if (isLightboxVisible) {
+      barrierRef.current?.focus();
+      lightbox.addEventListener('keydown', handleTabKeyPress);
+      lightbox.addEventListener('keydown', handleEscapeKeyPress);
+    }
+
+    return () => {
+      lightbox?.removeEventListener('keydown', handleTabKeyPress);
+      lightbox?.removeEventListener('keydown', handleEscapeKeyPress);
+    };
+  }, [isLightboxVisible, handleEscapeKeyPress]);
+
+  const imageTitle =
+    images[imageIndex].title || images[imageIndex].photographer;
+  const imagePhotogrpher = images[imageIndex].photographer;
+  const imageUrl = images[imageIndex].url;
+
+  const renderLightboxComponent = (): JSX.Element => (
+    <div
+      id={lightboxUid}
+      role="dialog"
+      aria-modal
+      aria-labelledby={`${lightboxUid}-title`}
+      className={styles.lightbox}
+      onClick={toggleLightbox}
+      ref={lightboxRef}
+    >
+      <div
+        onClick={(e) => {
+          e.stopPropagation();
+        }}
+        className={styles.lightboxContent}
+        onKeyDown={handleEscapeKeyPress}
+      >
+        <div className={styles.inner}>
+          <h2
+            id={`${lightboxUid}-title`}
+            className={styles.screenReaderText}
+            ref={barrierRef}
+            tabIndex={-1}
+          >
+            {imageTitle}
+          </h2>
+          <figure className={styles.imageCardWrapper} aria-label={imageTitle}>
+            <img alt={imageTitle} src={imageUrl} />
+          </figure>
+          <figcaption
+            className={classNames(styles.photographer, styles.withMargin)}
+          >
+            {imagePhotogrpher}
+          </figcaption>
+          <div className={styles.actionsWrapper}>
+            <Button
+              iconLeft={<IconAngleLeft />}
+              onClick={handlePreviousClick}
+              theme="black"
+              variant="secondary"
+            >
+              <span className={styles.screenReaderText}>{previous}</span>
+            </Button>
+            <Button
+              iconLeft={<IconAngleRight />}
+              onClick={handleNextClick}
+              theme="black"
+              variant="secondary"
+            >
+              <span className={styles.screenReaderText}>{next}</span>
+            </Button>
+          </div>
+          <button
+            className={styles.closeButton}
+            id={`close-${lightboxUid}`}
+            type="button"
+            aria-controls={lightboxUid.toString()}
+            aria-expanded="true"
+            onClick={toggleLightbox}
+          >
+            <span className={styles.screenReaderText}>
+              {closeButtonLabelText}
+            </span>
+            <svg viewBox="0 0 24 24" aria-hidden="true" tabIndex={-1} />
+          </button>
```

Comment:
I would still split these button sections to 2 different new components: "actions" and "closebutton": <Lightbox.actions/> and <Lightbox.closeButton/>. But that's just my opinion.

Reply:
I would keep them in same file still.